desk/lib/metagrab.hoon

```diff
@@ -0,0 +1,271 @@
+::  metagrab: parse metadata from html page
+::
+::    structured primarily for opengraph metadata, but supports arbitrary
+::    <meta> tag namespaces and many known <link> relationships.
+::
+/+  rh=re-html
+::
+|%
+::  searching for metadata
+::
++$  tope
+  $:  ns=@t     ::  property namespace, if any
+      key=@t    ::  property key
+      val=veal  ::  property value
+  ==
+::
++$  veal
+  $@  @t                 ::  flat property, or:
+  $:  top=@t             ::  property value
+      met=(map @t veal)  ::  w/ metadata
+  ==
```

Comment:
These type names are _garbage_ and I desperately need to replace them with something better.